astropy/cosmology/tests/test_cosmology.py

```diff
@@ -1582,6 +1582,46 @@ def test_z_at_value():
             z_at_value(cosmo.angular_diameter_distance, 1500*u.Mpc, zmin=4.)
 
 
+@pytest.mark.skipif('not HAS_SCIPY')
+@pytest.mark.parametrize('method', ['Brent', 'Golden', 'Bounded'])
+def test_z_at_value_bracketed(method):
```

Comment:
I think some high precision tests would also be valuable. All these are rtol=1e-6.

Reply:
I think I had made some experiments where the precision did not increase very smoothly with lowering `tol` in `minimize_scalar`, also many of the forward tests don't have higher precision either; so in light of the "different values on different architectures" in the comment it's unclear whether that can be achieved at all. Could try a test run on the different CI architectures perhaps. But rtol=1.e-6 comes down to a few 100 m/s in redshift at most, so this is already well above any precision relevant for practical use cases.

Reply:
Tried a bit higher precision now especially in the roundtrip tests; the comment did not really make sense to me especially with the first test that was actually roundtripping too.